src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2570 @@
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_pkcs11_test.c
+ * @brief Integration tests for the corePKCS11 implementation.
+ */
+/* Standard includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* corePKCS11 includes. */
+#include "core_pki_utils.h"
+#include "core_pkcs11.h"
+
+/* Device provisioning. */
+#include "aws_dev_mode_key_provisioning.h"
+
+/* Test includes. */
+#include "unity_fixture.h"
+#include "unity.h"
+
+/* mbedTLS includes. */
+#include "mbedtls/sha256.h"
+#include "mbedtls/pk.h"
+#include "mbedtls/pk_internal.h"
+#include "mbedtls/oid.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/entropy_poll.h"
+#include "mbedtls/x509_crt.h"
+
+/* corePKCS11 test includes. */
+#include "core_pkcs11_test.h"
+#include "test_execution_config.h"
+#include "test_param_config.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Number of simultaneous tasks for multithreaded tests.
+ *
+ * Each task consumes both stack and heap space, which may cause memory allocation
+ * failures if too many tasks are created.
+ */
+#define PKCS11_TEST_MULTI_THREAD_TASK_COUNT    ( 2 )
+
+/**
+ * @brief The number of iterations of the test that will run in multithread tests.
+ *
+ * A single iteration of Signing and Verifying may take up to a minute on some
+ * boards. Ensure that PKCS11_TEST_WAIT_THREAD_TIMEOUT_MS is long enough to accommodate
+ * all iterations of the loop.
+ */
+#define PKCS11_TEST_MULTI_THREAD_LOOP_COUNT    ( 10 )
+
+/**
+ * @brief All tasks of the SignVerifyRoundTrip_MultitaskLoop test must finish within
+ * this timeout, or the test will fail.
+ */
+#define PKCS11_TEST_WAIT_THREAD_TIMEOUT_MS     ( 1000000UL )
+
+/**
+ * @brief Make use of the unity TEST_PRINTF function to print log if supported.

```

Comment:
The test try to make use of unity log function to print debug log.
https://github.com/ThrowTheSwitch/Unity/blob/master/src/unity.h#L107-L109
The condition check is to prevent compile error if UNITY_INCLUDE_PRINT_FORMATTED is not enabled by developers.
The comments is updated in this PR to prevent misunderstanding.